compiler/noirc_frontend/src/elaborator/scope.rs

```diff
@@ -0,0 +1,112 @@
+use rustc_hash::FxHashMap as HashMap;
+
+use crate::hir::comptime::Value;
+use crate::{
+    hir::{
+        def_map::{ModuleDefId, TryFromModuleDefId},
+        resolution::errors::ResolverError,
+    },
+    hir_def::{
+        expr::{HirCapturedVar, HirIdent},
+        traits::Trait,
+    },
+    macros_api::{Path, StructId},
+    node_interner::{DefinitionId, TraitId, TypeAliasId},
+    Shared, StructType,
+};
+
+use super::Elaborator;
+
+#[derive(Default)]
+pub(super) struct Scope {
```

Comment:
I'm going to remove `Scope` entirely. This was made when I was reimplementing things before I started copying anything. Since the ScopeForest is used instead I'll update this to remove Scope.